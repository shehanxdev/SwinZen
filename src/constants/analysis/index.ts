export * from './analysis.enum';
export * from './pro-tips-content';
