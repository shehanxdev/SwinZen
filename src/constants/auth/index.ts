export * from './otp-data.enum';
