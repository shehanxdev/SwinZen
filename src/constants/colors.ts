/* eslint-disable @typescript-eslint/no-shadow */
/* eslint-disable @typescript-eslint/no-redeclare */

//NOTE:: Since an enum member cannot have a numeric name prefixed with "Sz"
enum Primary {
  Sz900 = `#070707`,
  Sz800 = `#0C1E0F`,
  Sz700 = `#16501D`,
  Sz600 = `#28642A`,
  Sz500 = `#45734A`,
  Sz400 = `#69AA67`,
  Sz300 = `#8DBE8B`,
  Sz200 = `#C0DBBF`,
  Sz100 = `#F0F7F0`, //NOTE::figma contain two different colors(#F0F7F0, #C0DBBF) with the border. #C0DBBF neeeds to be removed since it does not consider within the colors
}

enum Neutral {
  Sz1000 = `#283E36`, //NOTE::not available within the design system colors
  Sz900 = `#212529`,
  Sz800 = `#343A40`,
  Sz700 = `#495057`,
  Sz600 = `#6C757D`,
  Sz500 = `#ADB5BD`,
  Sz400 = `#CED4DA`,
  Sz300 = `#DEE2E6`,
  Sz200 = `#E9ECEF`,
  Sz100 = `#F8F9FA`, //NOTE::figma contain two different colors(#F8F9FA, #DEE2E6) with the border. #DEE2E6 neeeds to be removed since it does not consider within the colors
  Black = `#000000`,
  White = `#FFFFFF`,
}

enum Secondary {
  Sz900 = `#F65815`,
  Sz800 = `#F87944`,
  Sz700 = `#F99469`,
  Sz600 = `#FAA17C`,
  Sz500 = `#FBBCA1`,
  Sz400 = `#FDD7C7`,
  Sz300 = `#FDE4D9`,
  Sz200 = `#FFF2ED`,
  Sz100 = `#FFE3B0`, //NOTE::figma contain two different colors(##FFE3B0, #FFF9F6) with the border. #FFF9F6 neeeds to be removed since it does not consider within the colors
}

/* Currently this is available within the design system as a Secondary value. Since two different colors cannot exist in there, this has to be added to the design system */
enum Tertiary {
  Sz900 = `#A2FD2F`,
  Sz800 = `#B5FD59`,
  Sz700 = `#C4FE7A`,
  Sz600 = `#CBFE8A`,
  Sz500 = `#DAFEAC`,
  Sz400 = `#E9FFCD`,
  Sz300 = `#F0FEDE`,
  Sz200 = `#F8FFEF`,
  Sz100 = `#DAFEAC`, //NOTE::figma contain two different colors(#DAFEAC, #FBFFF7) with the border. #FBFFF7 neeeds to be removed since it does not consider within the colors
}

enum Transparency {
  Sz100 = `#212529`,
  Sz80 = `#212529cc`,
  Sz64 = `#212529a3`,
  Sz56 = `#2125298f`,
  Sz40 = `#21252966`,
  Sz24 = `#2125293d`,
  Sz16 = `#21252929`,
  Sz8 = `#21252914`,
  Sz4 = `#2125290a`,
  full = `#ffffff00`, //NOTE:: This is full transpapernt color. Not available under the design system.
}

enum Error {
  SzMain = `#E35F5F`, //Not defined within the design system but being used within the screens :(
}

const SwingZenThemePalette = {
  Primary,
  Neutral,
  Secondary,
  Tertiary,
  Transparency,
  Error,
};

export type Color = Primary | Neutral | Secondary | Tertiary | Transparency | Error;

export const Color = {
  ...SwingZenThemePalette,
};
