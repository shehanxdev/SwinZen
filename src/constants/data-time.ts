export const dateFormats = {
  default: 'YYYY-MM-DD',
  calenderHeader: 'MMMM YYYY',
  rangeDatePicker: 'MMM DD',
  scratchedDateAndTime: 'HH:MM, DD/MM/YY',
  singleDatePicker: 'ddd, MMM DD',
  textInput: 'DD / MM / YY',
  extrasMarketHeader: 'dd/MM/YY',
  finalResultsHeader: 'dd/MM/YYYY',
  dayAndDate: 'ddd, DD',
  timestamp: 'DD/MM/YYYY HH:mm:ssPM Z',
  advertisedStartTime: 'ddd, DD MMM YYYY HH:mm PM',
  saveFileTimestamp: 'DD_MM_YYYY_HH_mm_ss',
};
