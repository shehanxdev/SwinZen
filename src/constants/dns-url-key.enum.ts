//TODO::fill other dns url keys
export enum DnsUrlKey {
  Auth0 = 'auth0',
}
