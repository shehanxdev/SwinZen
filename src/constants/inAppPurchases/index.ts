export * from './plan-sku-data';
