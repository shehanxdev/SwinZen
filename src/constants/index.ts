export * from './auth';
export * from './error';
export * from './colors';
export * from './data-time';
export * from './navigation-routes';
export * from './regular-expressions';
export * from './runtime-environment';
export * from './static-info';
export * from './typography';
