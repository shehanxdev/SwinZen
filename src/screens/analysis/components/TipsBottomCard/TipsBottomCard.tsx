import React from 'react';
import { View } from 'react-native';

import { Button, Text } from '@sz/components';
import { tw } from '@sz/config';
import { Color, Route, ScoreType, TextAlignment, TextVariant } from '@sz/constants';
import { NavigationService } from '@sz/services';

export function TipsBottomCard({ type, route }) {
  const getTypeDescription = (type: ScoreType): string => {
    switch (type) {
      case 'Setup':
        return 'Shoulder tilt is too flat. Bring your lead shoulder up a bit more at address position.';
      case 'Backswing':
        return 'Hips are swaying in backswing. Hips should rotate without too much sway.';
      case 'Downswing':
        return 'As you move through downswing donâ€™t hang back on the back leg, transfer your weight to front foot.';
    }
  };

  const getRouteData = (route: Route) => {
    switch (route) {
      case 'pga-pro-tips':
        return {
          title: 'PGA Pro Tip',
          description: getTypeDescription(type),
          leftButtonTitle: 'AI pro tips',
          rightButtonTitle: 'Side-by-side',
          onLeftButtonPress: () => NavigationService.navigate(Route.AIProTips),
          onRightButtonPress: () => NavigationService.navigate(Route.SideBySide),
        };
      case 'ai-pro-tips':
        return {
          title: 'AI Pro Tip',
          description: getTypeDescription(type),
          leftButtonTitle: 'AI pro tips',
          rightButtonTitle: 'Side-by-side',
          onLeftButtonPress: () => NavigationService.navigate(Route.AIProTips),
          onRightButtonPress: () => NavigationService.navigate(Route.SideBySide),
        };
      case 'side-by-side':
        return {
          title: 'Side-By-Side',
          description: getTypeDescription(type),
          leftButtonTitle: 'AI pro tips',
          rightButtonTitle: 'Side-by-side',
          onLeftButtonPress: () => NavigationService.navigate(Route.AIProTips),
          onRightButtonPress: () => NavigationService.navigate(Route.SideBySide),
        };
    }
  };

  return (
    <View style={tw`pb-8.5 pt-8 px-2 rounded-t-2xl bg-Neutral-Sz900`}>
      <View style={tw`mx-4`}>
        <Text variant={TextVariant.SubTitle2SemiBold} color={Color.Neutral.Sz400} textAlign={TextAlignment.Left}>
          {getRouteData(route).title}
        </Text>
        <View style={tw`mt-3 mb-5`}>
          <Text variant={TextVariant.Body2Regular} color={Color.Neutral.Sz400} textAlign={TextAlignment.Left}>
            {getRouteData(route).description}
          </Text>
        </View>
      </View>
      <View style={tw`flex-row justify-between px-3.5 gap-2.5`}>
        <View style={tw`grow-1 basis-0`}>
          <Button
            title={getRouteData(route).leftButtonTitle}
            textColor={Color.Neutral.Black}
            backgroundColor={Color.Tertiary.Sz900}
            borderColor={Color.Tertiary.Sz900}
            onPress={getRouteData(route).onLeftButtonPress}
          />
        </View>
        <View style={tw`grow-1 basis-0`}>
          <Button
            title={getRouteData(route).rightButtonTitle}
            textColor={Color.Tertiary.Sz900}
            backgroundColor={Color.Transparency.full}
            borderColor={Color.Primary.Sz650}
            onPress={getRouteData(route).onRightButtonPress}
          />
        </View>
      </View>
    </View>
  );
}
