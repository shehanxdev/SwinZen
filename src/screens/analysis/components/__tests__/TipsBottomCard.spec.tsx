import * as React from 'react';

import { TipType } from '@sz/constants';
import { renderWithProviders } from '@sz/test-utils';

import { TipsBottomCard } from '../TipsBottomCard';

describe('TipsBottomCard component', () => {
  const testID = 'TipsBottomCardTestID';
  const mockFn = jest.fn();
  const getRenderedScreen = () =>
    renderWithProviders(
      <TipsBottomCard description="dummyDescription" tipType={TipType.PGA_PRO_TIPS} onSetTipType={mockFn} />,
    );

  it(`should render TipsBottomCard component correctly`, async () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the TipsBottomCard component via testID', () => {
    const { getByTestId } = getRenderedScreen();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeDefined();
  });
});
