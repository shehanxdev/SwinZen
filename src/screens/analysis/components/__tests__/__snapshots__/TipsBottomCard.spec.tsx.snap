// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TipsBottomCard component should render TipsBottomCard component correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#212529",
        "borderTopLeftRadius": 16,
        "borderTopRightRadius": 16,
        "paddingBottom": 34,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 32,
      }
    }
    testID="TipsBottomCardTestID"
  >
    <View
      style={
        {
          "marginLeft": 16,
          "marginRight": 16,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#CED4DA",
              "lineHeight": 28,
            },
            {
              "fontFamily": "SourceSansPro-SemiBold",
              "fontSize": 24,
              "fontWeight": "600",
              "textAlign": "left",
            },
          ]
        }
      >
        PGA Pro Tip
      </Text>
      <View
        style={
          {
            "marginBottom": 20,
            "marginTop": 12,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#CED4DA",
                "lineHeight": 24,
              },
              {
                "fontFamily": "SourceSansPro-Regular",
                "fontSize": 16,
                "fontWeight": "normal",
                "textAlign": "left",
              },
            ]
          }
        >
          dummyDescription
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 10,
          "justifyContent": "space-between",
          "paddingLeft": 14,
          "paddingRight": 14,
        }
      }
    >
      <View
        style={
          {
            "flexBasis": 0,
            "flexGrow": 1,
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            collapsable={false}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              {
                "backgroundColor": "#A2FD2F",
                "borderColor": "#A2FD2F",
                "borderRadius": 10,
                "borderStyle": "solid",
                "borderWidth": 1,
                "elevation": 0,
                "justifyContent": "center",
                "minWidth": "auto",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "width": "100%",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 10,
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    {
                      "height": 48,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "SourceSansPro-Regular",
                        "fontWeight": "normal",
                      },
                      {
                        "textAlign": "left",
                      },
                      [
                        {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        {
                          "marginHorizontal": 8,
                        },
                        {
                          "textTransform": "uppercase",
                        },
                        {
                          "color": "#6200ee",
                          "fontFamily": "SourceSansPro-SemiBold",
                          "fontWeight": "normal",
                        },
                        {
                          "fontFamily": "SourceSansPro-SemiBold",
                          "fontWeight": "normal",
                        },
                        {
                          "color": "#000000",
                        },
                      ],
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#000000",
                          "lineHeight": 28,
                        },
                        {
                          "fontFamily": "SourceSansPro-SemiBold",
                          "fontSize": 18,
                          "fontWeight": "bold",
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    AI pro tip
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "flexBasis": 0,
            "flexGrow": 1,
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            collapsable={false}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              {
                "backgroundColor": "#00000000",
                "borderColor": "#1A5C23",
                "borderRadius": 10,
                "borderStyle": "solid",
                "borderWidth": 1,
                "elevation": 0,
                "justifyContent": "center",
                "minWidth": "auto",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "width": "100%",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "disabled": false,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  {
                    "borderRadius": 10,
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    {
                      "height": 48,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#000000",
                        "fontFamily": "SourceSansPro-Regular",
                        "fontWeight": "normal",
                      },
                      {
                        "textAlign": "left",
                      },
                      [
                        {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        {
                          "marginHorizontal": 8,
                        },
                        {
                          "textTransform": "uppercase",
                        },
                        {
                          "color": "#6200ee",
                          "fontFamily": "SourceSansPro-SemiBold",
                          "fontWeight": "normal",
                        },
                        {
                          "fontFamily": "SourceSansPro-SemiBold",
                          "fontWeight": "normal",
                        },
                        {
                          "color": "#A2FD2F",
                        },
                      ],
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#A2FD2F",
                          "lineHeight": 28,
                        },
                        {
                          "fontFamily": "SourceSansPro-SemiBold",
                          "fontSize": 18,
                          "fontWeight": "bold",
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Side-by-side
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
